hibernate-core/src/main/java/org/hibernate/event/spi/EventType.java

```diff
@@ -76,6 +78,41 @@
 	public static final EventType<PostCollectionRemoveEventListener> POST_COLLECTION_REMOVE = create( "post-collection-remove", PostCollectionRemoveEventListener.class );
 	public static final EventType<PostCollectionUpdateEventListener> POST_COLLECTION_UPDATE = create( "post-collection-update", PostCollectionUpdateEventListener.class );
 
+	/**
+	 * Add a new event type.
+	 *
+	 * @param name - name of the custom event
+	 * @param listenerClass - the base listener class or interface associated with the entity type
+	 * @param <T> - listenerClass
+	 * @return the custom {@link EventType}
+	 */
+	public synchronized static <T> EventType<T> addCustomEventType(String name, Class<T> listenerClass) {
+		if ( name == null || listenerClass == null ) {
+			throw new HibernateException( "Custom EventType name and associated class must be non-null." );
+		}
+		final EventType existingEventType = EVENT_TYPE_BY_NAME_MAP.get( name );
+		if ( existingEventType != null ) {
+			if ( listenerClass.equals( existingEventType.baseListenerInterface ) ) {
+				LOG.warn( "EventType [" + name + "] with listener Class ["
```

Comment:
If we log this as a warning, how can we prevent Hibernate RX to log such warnings if it's started multiple times?

We would need to either have a way to check if such types have not been registered already, or register them statically - so making sure it happens once and once only.

Reply:
@Sanne, this check was removed in my most recent commit.